File: src/main/java/pl/oczadly/kotlin_app/hackyeah2025/prompt/PromptController.java
package pl.oczadly.kotlin_app.hackyeah2025.prompt;

import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/prompts")
public class PromptController {

    private final PromptRepository promptRepository;

    public PromptController(PromptRepository promptRepository) {
        this.promptRepository = promptRepository;
    }

    @PostMapping
    public void addPrompt(@RequestBody PromptDTO promptDTO) {
        Prompt prompt = new Prompt(UUID.randomUUID(), promptDTO.prompt(), promptDTO.response());
        promptRepository.save(prompt);
    }

    @GetMapping
    public List<Prompt> getAllPrompts() {
        return promptRepository.findAll();
    }
}

File: src/main/kotlin/pl/oczadly/kotlin_app/hackyeah2025/prompt/PromptController.kt
package pl.oczadly.kotlin_app.hackyeah2025.prompt

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/prompts")
class PromptController(private val promptRepository: PromptRepository) {

    @PostMapping
    fun addPrompt(@RequestBody promptDTO: PromptDTO) {
        val prompt = Prompt(prompt = promptDTO.prompt, response = promptDTO.response)
        promptRepository.save(prompt)
    }

    @GetMapping
    fun getAllPrompts() = promptRepository.findAll()
}